### PR TITLE
feat: add pattern commands plugin

### PR DESCRIPTION
**Pull request by feralheart**
With this plugin users can create a new command type what triggers if someone says a specific word

### CODE REVIEW

**Review by kyranet — CHANGES_REQUESTED** [review]
I don't agree with using "reactive" for this, gets confusing with [RxJS](https://www.npmjs.com/package/rxjs) as well as the reactivity concept Vue, React, and others have.

File: packages/reactive-commands/src/index.ts
Comment (favna) [inline packages/reactive-commands/src/index.ts:1]: We use `/register` for these kind of things, just look at other plugins.

**Review by favna — CHANGES_REQUESTED** [review]

File: packages/reactive-commands/README.md
Comment (favna) [inline packages/reactive-commands/README.md:47]:
```suggestion
import '@sapphire/plugin-reactive-commands/register';
```

**Review by favna — CHANGES_REQUESTED** [review]
register stuff as well. I'll review more thoroughly later when this and the name are resolved.

File: packages/pattern-commands/README.md
Comment (favna) [inline packages/pattern-commands/README.md:22]:
```suggestion
-   Starting April 2022 this plugin will only work with Message Intent. This is because this plugin relies on scanning messages, which is impossible without the intent.
```

**Review by favna — CHANGES_REQUESTED** [review]
btw GH shows the previous comment is outdated but it's not, you still need to do the whole /register stuff @feralheart

**Review by c43721 — DISMISSED** [review]

**Comment by c43721** [comment]
Question: What's the motivation to allow only strings for regex? Could this be changed in a future version?

(not that it really matter though)

**Comment by kaname-png** [comment]
> Question: What's the motivation to allow only strings for regex? Could this be changed in a future version?
> 
> (not that it really matter though)

Because as the plugin says, patterns, not plain text messages.

**Comment by favna** [comment]
> Question: What's the motivation to allow only strings for regex? Could this be changed in a future version?

What else would it be. RegEx are always strings. Don't mistake strings for A-Z exclusively here, you can do anything that is a valid RegEx. What it means that the RegEx has to be a string is that it has to be `aliases: ['a{3,}']` and not `aliases: [/a{3,}]` because `aliases` has to be a `string[]` as per the Framework requirements of that option.

**Comment by c43721** [comment]
Thanks for clearing it up. I had a custom regex plugin to use regexes, didn't really think twice about it.

**Review by favna — APPROVED** [review]

**Comment by favna** [comment]
Blocked by #199 because of how Lerna versions packages I need the repo to be completely cleanly versioned before merging this so I can do a force major to 1.0.0 for this 🙄

**Pull request merged**
